tests/test_embed.rb

```diff
@@ -0,0 +1,184 @@
+require 'dogapi'
+require 'time'
+require 'test_base.rb'
+
+class TestEmbed < Test::Unit::TestCase
+  include TestBase
+
+  def test_get_all_embeds
+    dog = Dogapi::Client.new(@api_key, @app_key)
+
+    # Attempt API Call
+    status, result = dog.get_all_embeds()
+    # Sanity check results
+    assert_equal status, "200", "invalid HTTP response => #{status}"
+    assert result.has_key?("embedded_graphs")
+  end
+
+  def test_create_embed
+    dog = Dogapi::Client.new(@api_key, @app_key)
+    # create an embed using the embed id and verify that it is valid
+    graph_json = '{
+      "viz": "toplist",
+      "requests": [{
+        "q": "top(system.disk.free{$var} by {device}, 10, \'mean\', \'desc\')",
+        "style": {
+          "palette": "dog_classic"
+        },
+        "conditional_formats": [{
+          "palette": "red",
+          "comparator": ">",
+          "value": 50000000000
+        }, {
+          "palette": "green",
+          "comparator": ">",
+          "value": 30000000000
+        }]
+      }]
+    }'
+    timeframe = "1_hour"
+    size = "medium"
+    legend = "no"
+    title = "Sick Title!"
+    description = {
+      :timeframe => timeframe,
+      :size => size,
+      :legend => legend,
+      :title => title
+    }
+    status, result = dog.create_embed(graph_json, description)
+    # Sanity check results
+    assert result["graph_title"] == title
+    assert_equal status, "200", "invalid HTTP response => #{status}"
```

Comment:
you can but you would have to parse the html though which is kind of annoying. but I can add it if you think it's worth adding